### DISCUSSION TITLE
pyzo shell not load  file  “.pdbrc”

### DISCUSSION THREAD

**Discussion by wilikeit** [pyzo/pyzo#1080] [Q&A]
~/.pdbrc
import pprint 
alias pp pprint.pprint(%1,width=1)

I found that in pyzo, it  will not  load this file ( ~/.pdbrc), but use  "python3 script.py" to debug  will load this file, is that how pyzo is designed ?

**Comment by bdieterm** [comment]
The debug commands in Pyzo are different than the ones in pdb. Therefore ".pdbrc" files are not supported.
Debug commands in Pyzo can be executed via the "Shell" menu, the shell toolbar, by setting breakpoints in the editor and by executing magic commands. Type `db help` in the shell to see a list of all magic debug commands. For example: `db continue` will continue execution.

**Reply by wilikeit** [reply]
> The debug commands in Pyzo are different than the ones in pdb. Therefore ".pdbrc" files are not supported. Debug commands in Pyzo can be executed via the "Shell" menu, the shell toolbar, by setting breakpoints in the editor and by executing magic commands. Type `db help` in the shell to see a list of all magic debug commands. For example: `db continue` will continue execution.


 Can't pyzo use pp command, or is there an alternative command ？

**Reply by bdieterm** [reply]
According to your first message, `pp` is an alias for `pprint.pprint`, defined in "~/.pdbrc".
You can use that in every Python interpreter, also with Pyzo. For example:
```python3
import pprint; pp = pprint.PrettyPrinter(indent=4, width=40, sort_dicts=False).pprint

# example for using pp:
myvar = list(range(0, 10000, 1000))
pp(myvar)
```

**Reply by wilikeit** [reply]
Well, that's a solution
